unified_planning/transformers/negative_conditions_remover.py

```diff
@@ -147,6 +147,8 @@ def get_rewritten_problem(self) -> Problem:
 
         for action in self._problem.actions:
             if isinstance(action, InstantaneousAction):
+                if action.simulated_effects is not None:
+                    raise up.exceptions.UPUsageError('NegativeConditionsRemover does not work with simulated effects')
```

Comment:
can't we trap this with the `ProblemKind` at the beginning of the function?

Reply:
Right!